TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/middleware/clusters.py

```diff
@@ -66,72 +66,8 @@ def clear_indices(cluster: Cluster):
-
-    :param cluster: Cluster object to interact with the Elasticsearch cluster.
-    :param repository: Name of the snapshot repository to clear snapshots from.
-    :raises Exception: For general errors during snapshot clearing, except when the repository is missing.
-    """
-    try:
-        # List all snapshots in the repository
-        snapshots_path = f"/_snapshot/{repository}/_all"
-        response = call_api(cluster, snapshots_path, raise_error=True)
-        logger.debug(f"Raw response: {response.json()}")
-        snapshots = response.json().get("snapshots", [])
-        logger.info(f"Found {len(snapshots)} snapshots in repository '{repository}'.")
-
-        if not snapshots:
-            logger.info(f"No snapshots found in repository '{repository}'.")
-            return
-
-        # Delete each snapshot
-        for snapshot in snapshots:
-            snapshot_name = snapshot["snapshot"]
-            delete_path = f"/_snapshot/{repository}/{snapshot_name}"
-            call_api(cluster, delete_path, method=HttpMethod.DELETE, raise_error=True)
-            logger.info(f"Deleted snapshot: {snapshot_name} from repository '{repository}'.")
-
-    except Exception as e:
-        # Handle 404 errors specifically for missing repository
-        if isinstance(e, HTTPError) and e.response.status_code == 404:
-            error_details = e.response.json().get('error', {})
-            if error_details.get('type') == 'repository_missing_exception':
-                logger.info(f"Repository '{repository}' is missing. Skipping snapshot clearing.")
-                return
-        # Re-raise other errors
-        logger.error(f"Error clearing snapshots from repository '{repository}': {e}")
-        raise e
-
-
-def delete_repo(cluster: Cluster, repository: str) -> None:
-    logger.info(f"Deleting repository '{repository}'")
-    """
-    Delete repository. Should be empty before execution.
-
-    :param cluster: Cluster object to interact with the Elasticsearch cluster.
-    :param repository: Name of the snapshot repository to delete.
-    :raises Exception: For general errors during repository deleting, except when the repository is missing.
-    """
-    try:
-        delete_path = f"/_snapshot/{repository}"
-        call_api(cluster, delete_path, method=HttpMethod.DELETE, raise_error=True)
-        logger.info(f"Deleted repository: {repository}.")
-    except Exception as e:
-        # Handle 404 errors specifically for missing repository
-        if isinstance(e, HTTPError) and e.response.status_code == 404:
-            error_details = e.response.json().get('error', {})
-            if error_details.get('type') == 'repository_missing_exception':
-                logger.info(f"Repository '{repository}' is missing. Skipping delete.")
-                return
-        # Re-raise other errors
-        logger.error(f"Error deleting repository '{repository}': {e}")
-        raise e
+    if snapshot:
+        snapshot.delete_all_snapshots()
```

Comment:
Snapshot should be named snapshot repo?  I doesn't make much sense to delete all snapshots of a snapshot

Reply:
Our Snapshot class has definitely grown over time to encompass all things snapshot related. We may want to refactor this to something like SnapshotManager as thinking of this as a single snapshot will cause confusion